components/clp-package-utils/clp_package_utils/general.py

```
@@ -574,3 +577,21 @@ def validate_path_for_container_mount(path: pathlib.Path) -> None:
+
+def validate_dataset(clp_config: CLPConfig, dataset: str) -> None:
+    """
+    Checks if the provided dataset currently exists in the metadata database.
+    :param clp_config:
+    :param dataset:
+    """
+    db_config: Database = clp_config.database
+    sql_adapter: SQL_Adapter = SQL_Adapter(db_config)
+    clp_db_connection_params: dict[str, any] = db_config.get_clp_connection_params_and_type(True)
+    table_prefix: str = clp_db_connection_params["table_prefix"]
+    with closing(sql_adapter.create_connection(True)) as db_conn, closing(
```

Flake8 (7.2.0)

[error] undefined name 'closing'

(F821)